include/layers/PoolingLayer.hpp

```diff
@@ -0,0 +1,136 @@
+#pragma once
+#include <algorithm>
+#include <cstdlib>
+#include <string>
+
+#include "layers/Layer.hpp"
+
+template <typename ValueType>
+ValueType avg_pooling(const std::vector<ValueType>& input) {
+  if (input.empty()) {
+    throw std::runtime_error("Empty input in avg pooling");
+  }
+  return std::accumulate(input.begin(), input.end(), ValueType(0)) /
+         input.size();
+}
+
+template <typename ValueType>
+ValueType max_pooling(const std::vector<ValueType>& input) {
+  if (input.empty()) {
+    throw std::runtime_error("Empty input in max pooling");
+  }
+  return *(std::max_element(input.begin(), input.end()));
+}
+
+template <typename ValueType>
+class PoolingLayer : public Layer<ValueType> {
+ public:
+  PoolingLayer() = delete;
+  PoolingLayer(const Shape& input_shape, const Shape& pooling_shape,
+               const std::string& pooling_type = "average");
+  PoolingLayer(const PoolingLayer& c) = default;
+  PoolingLayer& operator=(const PoolingLayer& c) = default;
+  std::vector<ValueType> run(const std::vector<ValueType>& input) const;
+
+ private:
+  Shape poolingShape_;
+  char poolingType_;
+};
+
+template <typename ValueType>
+PoolingLayer<ValueType>::PoolingLayer(const Shape& input_shape,
+                                      const Shape& pooling_shape,
+                                      const std::string& pooling_type)
+    : poolingShape_(pooling_shape), Layer<ValueType>(input_shape, input_shape) {
+  if (pooling_shape.dims() > input_shape.dims()) {
+    throw std::invalid_argument("Pooling dims is bigger than the input dims");
+  }
+  if (pooling_shape.dims() > 2 || pooling_shape.dims() == 0) {
+    throw std::invalid_argument("Pooling dims is bigger than 2 or is 0");
+  }
+  if (pooling_type == "average") {
+    poolingType_ = 0;
+  } else if (pooling_type == "max") {
+    poolingType_ = 1;
```

Comment:
Introduce and use enum here for pooling type

Reply:
Ok